src/wasm-lib/kcl/src/execution/mod.rs

```diff
@@ -776,6 +776,12 @@ impl ExecutorContext {
             )
             .await;
 
+        // Flush the batch with all the end commands.
+        // If we don't do this then some programs might not fully finish executing.
+        self.engine
+            .flush_batch(true, SourceRange::new(program.end, program.end, ModuleId::default()))
+            .await?;
```

Comment:
This should not early return. If there's an error flushing the final batch, the `?` will make it return to the UI without building the graph. The whole point of this function is to capture the `Result` of execution, build the graph, and return the result with the graph.

Reply:
ya i was wondering how to do this and ensure it works w the other stuff

Reply:
i actually have a different idea i see the issue we call this in exec_ast which is fine but like then we set scene units after that and we dont need to, anyways i ahve a better fix

Reply:
okay pushed way better solution

Reply:
i got to the bottom of the real problem :)

Reply:
no im going to add another test just to ensure another way we dont revert

Reply:
So we don't flush the batch? That's okay?

Reply:
we do! its in exec_ast at the end of the module, so that WAS working up until this PR which added an extra SetSceneUnits on to the end of executing everytjing (https://github.com/KittyCAD/modeling-app/pull/5529), which then was not included in the batch . this was an unintentional after effect of that PR that i dont think was intended. So basically, we need to just ensure if we are at the end of the Main module (meaning we've executed everything else) that we don't append another to the batch :) and i added a test case for ensuring the batch does not get appending and stays empty too :) and since the other artifact graphs dont change now we also know all the tests for multi-module chaning of units are safe anyways i feel like 80000x more confident in this PR now

Reply:
But I wanted `cube_with_error` to change.

That test is a copy of the `cube` test with an error after it. The batch isn't getting flush on `main`. `cube_with_error`'s artifact_commands are truncated compared to `cube`.

Reply:
anyways long story is i was going crazy because i had tests from like 6 months ensuring we flush on batch like specific kcl files and i was like if this broke why didnt those tests break, well they were testing specific kcl code NOT that batch was empty AND the batch was still empty since we just moved the flush to the end of modules. Its just set scene units that is fucking up

Reply:
I guess that only happens when there's an error in execution, so maybe it doesn't matter.

Reply:
anyways now we have 3 more fail safe tests :)

Reply:
so i think cube_with_error is performing well because we shouldnt execute the rest of the batch if something fails :) i think its doing the right thing and stopping